packages/SystemUI/src/com/android/systemui/keyguard/KeyguardViewMediator.java

```diff
@@ -1409,6 +1433,10 @@ private void doKeyguardLocked(Bundle options) {
 
         if (DEBUG) Log.d(TAG, "doKeyguard: showing the lock screen");
         showLocked(options);
+        final int rebootAfterTimeout = Settings.Global.getInt(mContext.getContentResolver(), "settings_reboot_after_timeout", 0);
+        if (rebootAfterTimeout != 0) {
+            doRebootForOwnerAfterTimeoutIfEnabled(rebootAfterTimeout);
+        }
```

Comment:
Because of the nature of this feature, it feels like it would be good if we hardcoded a lower bound like 1 hour  (which is the lowest value of the UI implementation)

Reply:
i do not want to harcode a default value. this feature is disabled by default

Reply:
I'm not talking about a default value; I'm talking about a hardcoded guard, like `if (rebootAfterTimeout >= 1 hour)`

Reply:
we don't really support users changing these values through adb or other unsupported methods. that would be completely on them if they messed up

Reply:
anyway it seems 2 minutes is good enough for the guard value